test/sequential/sequencerStaking.cancel.test.ts

```diff
@@ -1,403 +1,281 @@
-  let reqIdCanceled: number = 0;
-  if (cancellation === true) {
-    const cancel = await Sudo.asSudoFinalized(
-      Sudo.sudoAsWithAddressString(
-        cancelerAddress,
-        await Rolldown.cancelRequestFromL1(chain, reqId),
-      ),
-    );
-    await waitSudoOperationSuccess(cancel, "SudoAsDone");
-    reqIdCanceled = Rolldown.getRequestIdFromCancelEvent(cancel);
-  }
-  return { txIndex, api, reqId, reqIdCanceled };
-}
-
-const preSetupSequencers = {
-  Ethereum: "0x3cd0a705a2dc65e5b1e1205896baa2be8a07c6e0",
-  Arbitrum: "0x798d4ba9baf0064ec19eb4f0a1a45785ae9d6dfc",
-};
+let chain: any;
+let testUser1: User;
+let testUser2: User;
+let testUser2Address: string;
+let disputePeriodLength: number;
+let providingExtrinsic: any;
 
 beforeAll(async () => {
   await initApi();
-  setupUsers();
   await setupApi();
-  //Add a few tokes because some tests may end up on slashing them
-  await Sudo.batchAsSudoFinalized(
-    Sudo.sudoAsWithAddressString(
-      preSetupSequencers.Ethereum,
-      await SequencerStaking.provideSequencerStaking(BN_ZERO, "Ethereum"),
-    ),
-    Sudo.sudoAsWithAddressString(
-      preSetupSequencers.Arbitrum,
-      await SequencerStaking.provideSequencerStaking(BN_ZERO, "Arbitrum"),
-    ),
-  );
+  disputePeriodLength = (await Rolldown.disputePeriodLength()).toNumber();
 });
 
 beforeEach(async () => {
-  //TODO: Replace this by some monitoring of the active queue.
-  await waitForNBlocks((await Rolldown.disputePeriodLength()).toNumber());
+  //There shouldn't be any sequencer in activeSequencers
+  [testUser1, testUser2] = setupUsers();
+  await Sudo.batchAsSudoFinalized(
+    Assets.mintNative(testUser1),
+    Assets.mintNative(testUser2),
+  );
   const activeSequencers = await SequencerStaking.activeSequencers();
-  let anySequencerGone = false;
   for (const chain in activeSequencers.toHuman()) {
     for (const seq of activeSequencers.toHuman()[chain] as string[]) {
-      if (
-        seq !== preSetupSequencers.Ethereum &&
-        seq !== preSetupSequencers.Arbitrum
-      ) {
-        await leaveSequencingIfAlreadySequencer(seq);
-        anySequencerGone = true;
+      if (seq !== null) {
+        await leaveSequencing(seq);
       }
     }
   }
-  if (anySequencerGone) {
-    await waitForNBlocks(10);
-  }
+  chain = "Ethereum";
+  const minToBeSequencer = await SequencerStaking.minimalStakeAmount();
+  providingExtrinsic = await SequencerStaking.provideSequencerStaking(
+    minToBeSequencer.addn(1000),
+    "Ethereum",
+  );
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAs(testUser1, providingExtrinsic),
+    Sudo.sudoAs(testUser2, providingExtrinsic),
+  );
+  const sequencers = await SequencerStaking.activeSequencers();
+  expect(sequencers.toHuman().Ethereum).toContain(
+    testUser1.keyRingPair.address,
+  );
+  expect(sequencers.toHuman().Ethereum).toContain(
+    testUser2.keyRingPair.address,
+  );
+  testUser2Address = testUser2.ethAddress.toString();
```

Comment:
use the keyring. address instead, I run into soo many problems with that ethAddress. I am about to remove it soon

Reply:
Done